### ISSUE TITLE
Testing / Command line questions

### ISSUE THREAD

**Issue by fmr** [tonylukasavage/ti-mocha#29]
Hi @tonylukasavage,

I've been looking into testing within the Appcelerator ecosystem, and I always seem to come back to one of your projects, namely ti-mocha.

So, I have a few questions:
- How does one set up tests for an Alloy project using ti-mocha?
- Is there a way to run tests via command line? (I didn't see any commands in the [example](http://tonylukasavage.com/ti-mocha)).
- Is there a difference between test suites testing views and ones that test a library?

My goal is to have an alloy project set up and have tests running on travis-ci. Likewise, I want to create reusable modules installable using gittio, also built and tested on travis-ci.

Any help pointing me in the right direction would be much appreciated. Thanks in advance, and thanks for all your awesome work.

**Comment by yuchi** [comment]
> My goal is to have an alloy project set up and have **tests running on travis-ci**.

(emphasis by me)

I cannot really help you on the other questions, but I can point you to [this `.travis.yml`](https://github.com/smclab/ti-superagent/blob/master/.travis.yml) that installs all the required dependencies and launches an app in the Travis CI environment.

**Comment by fmr** [comment]
Thanks @yuchi, that's pretty helpful. I'd love to have some automated tests there, but at the very least with your example I'll know that the project builds fine on Travis.

**Comment by yuchi** [comment]
Please note that the configuration I linked is iOS only.

I [built an experiment](https://github.com/yuchi/ti-cross-ci-experiments/blob/master/.travis.yml) that makes tests running in both iOS and Android, both on Travis CI. You can see the resulting runs here: https://travis-ci.org/yuchi/ti-cross-ci-experiments

**Comment by yuchi** [comment]
@fmr, could you please add a few notes on how you structured your project in the end?

**Comment by fmr** [comment]
@yuchi, In the end I didn't end up using ti-mocha (or any test framework for that matter, sadly). I know very little of the best practices for testing in the javascript world, much less in Titanium.

What I did manage to get working was your ti-cross-ci-experiments set up. I haven't got any tests, but at least any glaring build errors will come up on Travis for iOS and Android.

I've created a gist for what I've used: https://gist.github.com/fmr/430c45eb93a000ec1f42
A successful build looks like this:

![image](https://cloud.githubusercontent.com/assets/592174/8793376/1d687904-2fb9-11e5-804c-67d14a362c57.png)

I would still love to know how I can integrate ti-mocha or other test frameworks in a mobile app project (not just for commonjs modules, etc.) for use with continuous integration.
